README.md

```diff
@@ -18,18 +18,45 @@ The GPU Indice Generation algorithm is a unofficial implementation of paper [SEC
     1. In Windows use the [./conda/spconv-windows.yml](./conda/spconv-windows.yml) when creating the environment
     2. In Ubuntu use the [./conda/spconv-linux.yml](./conda/spconv-linux.yml) when creating the environment
 4. Install [CMake](https://apt.kitware.com/) and [git](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git) (used by CMake to fetch content)
-5. Install a C++14 (or higher) compatible compiler. 
-    1. In Ubuntu you can install `build-essential` package
+5. Install a C++14 (or higher) compatible compiler. It must be compatible with the installed CUDA version (some [compatibilities](https://gist.github.com/ax3l/9489132)).
+    1. In Ubuntu you can install `build-essential` package<br>
+    If a specific version of g++ and gcc is needed, follow this instructions for a Linux based system (from [stackoverflow](https://askubuntu.com/questions/26498/how-to-choose-the-default-gcc-and-g-version)).
+    ```sh
+    # First remove update-alternatives for gcc and g++
+    sudo update-alternatives --remove-all gcc 
+    sudo update-alternatives --remove-all g++
+    # Install required gcc and g++ packages (e.g. gcc and g++ 11)
+    sudo apt-get install gcc-11 g++-11
+    # Install alternatives
+    # sudo update-alternatives --install <link> <name> <path> <priority>
+    sudo update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-11 10
+
+    sudo update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-11 10
+
+    sudo update-alternatives --install /usr/bin/cc cc /usr/bin/gcc 30
+    sudo update-alternatives --set cc /usr/bin/gcc
+
+    sudo update-alternatives --install /usr/bin/c++ c++ /usr/bin/g++ 30
+    sudo update-alternatives --set c++ /usr/bin/g++
+
+    # If multiple alternatives are installed, configuration of the default commands for gcc and g++ can be done interactively
+    sudo update-alternatives --config gcc
+    sudo update-alternatives --config g++
+    ```
     2. In Windows you need to install Visual Studio version that is compatible with CUDA 11.7
+5. If `Boost` is not found in the system, CMake tries to fetch a header only version. If this fails, you need to install `Boost`.
```

Comment:
If spconv is being built inside a Conda environment, then it uses Boost from that environment (if found)...as is seen here:
```
# If a Conda environment is enabled, we look for the CUDA toolkit in that environment
if(DEFINED ENV{CONDA_PREFIX})
    message(STATUS "CONDA environment detected, path: $ENV{CONDA_PREFIX}")
    set(CUDAToolkit_ROOT "$ENV{CONDA_PREFIX}" CACHE PATH "Path to CUDA toolkit")
    set(CUDA_TOOLKIT_ROOT_DIR $ENV{CONDA_PREFIX} CACHE PATH "CUDA Toolkit directory path (used by older/other find_package implementations)")
    # Some Caffe2 find-package(s) need these if we want to find cudnn in Windows
    set(CUDNN_LIBRARY "$ENV{CONDA_PREFIX};$ENV{CONDA_PREFIX}/Library" CACHE PATH "CUDNN directory path (used by some Caffe2 find-packages)")
    set(CUDNN_INCLUDE_DIR "$ENV{CONDA_PREFIX};$ENV{CONDA_PREFIX}/Library" CACHE PATH "CUDNN directory path (used by some Caffe2 find-packages)")
    set(NVTOOLEXT_HOME $ENV{CONDA_PREFIX} CACHE PATH "Some pre-built PyTorch versions require this")
    # Boost should be installed in the environment
    list(APPEND CMAKE_PREFIX_PATH "$ENV{CONDA_PREFIX}/Library/lib/cmake/Boost-1.85.0")
endif()
```

Perhaps you can mention that on this line where you talk about `Boost`.

Reply:
A bit strange way to fetch boost library with a specific verison and inside cmake, I added this behavior in the doc as it could lead to some unexpected behavior. I also added ways to install cmake and boost through conda (as it seems it's expected).

Reply:
It's not being fetched anymore. If a Conda environment is activated, it will try to look for Boost in that directory. The reason it was being fetched using CMake was to simplify the build process, especially since if header-only part of Boost was being used, but making this fully work would be too complicated. 

This is what the Conda environment (for Linux) looks like currently:
```
name: spconv
channels:
    - pytorch
    - nvidia
    - conda-forge
    - anaconda
dependencies:
    - python=3.10
    - pytorch==2.0.1
    - torchvision=0.15.2
    - pytorch-cuda=11.7
    - cudatoolkit-dev=11.7
    - cudnn=8.9
    - nvtx
    - boost=1.85.0
    - mkl=2024.0
    - pip
    - pip:
        - opencv-python
        - build
        - numpy
```